src/thread_pool_pppcsr/thread_pool_pppcsr.cpp

```diff
@@ -0,0 +1,110 @@
+//
+// Created by Christian Menges.
+//
+
+#include "thread_pool_pppcsr.h"
+#include <vector>
+#include <thread>
+#include <queue>
+#include <mutex>
+#include <iostream>
+#include <ctime>
+#include <numa.h>
+#include <cmath>
+
+using namespace std;
+
+/**
+ * Initializes a pool of threads. Every thread has its own task queue.
```

Comment:
For now this desing is good, but we should re-evaluate whether having a single global task queue would be more beneficial.